File: MvpWithDagger/app/src/main/java/dev/eder/mvpwithdagger/di/AppComponent.java
package dev.eder.mvpwithdagger.di;

import javax.inject.Singleton;

import dagger.Component;
import dev.eder.mvpwithdagger.ui.login.LoginActivity;
import dev.eder.mvpwithdagger.ui.profile.ProfileActivity;
import dev.eder.mvpwithdagger.ui.ws.WsActivity;

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {

    void inject(LoginActivity loginActivity);

    void inject(ProfileActivity profileActivity);

    void inject(WsActivity wsActivity);

}

File: DaggerKotlinRetrofit/app/src/main/java/dev/eder/daggerkotlinretrofit/di/component/AppComponent.kt
package dev.eder.daggerkotlinretrofit.di.component

import android.content.Context
import dagger.Component
import dev.eder.daggerkotlinretrofit.di.modules.AppModule
import dev.eder.daggerkotlinretrofit.ui.MainActivity
import javax.inject.Singleton


@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {

    fun context(): Context

    fun inject(mainActivity: MainActivity)

}